adapter/config/types.go

```diff
@@ -125,6 +126,23 @@ type connection struct {
 	Timeouts connectionTimeouts
 }
 
+type rateLimit struct {
+	Enabled                  bool
+	Hostname                 string
+	Port                     uint32
+	FailureModeDeny          bool
+	RatelimitRequestTimeouts ratelimitRequestTimeouts
+	KeyFilePath              string
+	CertFilePath             string
+	CaCertFilePath           string
+	SSLCertSANHostname       string
+}
+
+type ratelimitRequestTimeouts struct {
+	RequestTimeoutInSeconds     int64
```

Comment:
What if we have a single field (in milli-seconds) and once we get the value we break down like this since envoy requires it in that manner.
It feels like this makes the config bit confusing.

Reply:
My bad, I think I suggest this.
+1, Let's follow the same way we follow other configs

Reply:
fixed with: https://github.com/wso2/product-microgateway/pull/3142/commits/c77b3dced5fc5331d2b5157ba174af38261d2fae